website/docs/feature-flag-tutorials/use-cases/ai.md

```diff
@@ -0,0 +1,187 @@
+-   `prompt`: The prompt to give to the LLM.
+
+Because all these models vary a lot in their capabilities, chances are you'll be testing multiple models, and multiple versions of the same model, and multiple prompts.
+
+This is where feature flags are super useful. And this is what we'll be doing in this tutorial.
+
+We'll be starting with a basic chatbot. I know building a chatbot is a total cliché at this point, but it's a small, self-contained example that everyone knows how to interact with. You can take this example and apply it to any other use case where you need to interact with an LLM via an API.
+
+First, clone the repo:
+
+```sh
+git clone https://github.com/alvinometric/feature-flags-ai
+cd feature-flags-ai
+```
+
+Install the dependencies
+
+```sh
+npm install
+```
+
+Run the app
+
+```sh
+npm run dev
+```
+
+And you should see something like this:
+
+![chatbot UI](./sveltekit-chatbot.png)
+
+This is a simple Sveltekit app with a chatbot UI. It doesn't really matter if you've never used Sveltekit before. It's similar to Next.js or Nuxt or SolidStart, where your file structure defines your routes.
+
+The most important file for this tutorial is the `src/routes/api/chat/+server.js` file.
+
+It creates an API endpoint at `/api/chat`. When your frontend sends a POST request to `/api/chat`, this is the code that handles it.
+
+```javascript
+import { createOpenAI } from "@ai-sdk/openai";
+import { streamText } from "ai";
+import { env } from "$env/dynamic/private";
+
+const openai = createOpenAI({
+    apiKey: env.OPENAI_API_KEY ?? "",
+});
+
+const variants = {
+    variant1: {
+        model: "gpt-3.5-turbo",
+        max_tokens: 4096,
+        temperature: 1,
+    },
+    variant2: {
+        model: "gpt-4-turbo",
+        max_tokens: 4096,
+        temperature: 1,
+    },
+    variant3: {
+        model: "gpt-4-vision-preview",
+        max_tokens: 4096,
+        temperature: 1,
+    },
+};
+
+export const POST = async ({ request }) => {
+    const { messages } = await request.json();
+    const variant = variants["variant1"];
+    const result = await streamText({
+        model: openai(variant.model),
+        messages,
+        max_tokens: variant.max_tokens,
+        temperature: variant.temperature,
+    });
+    return result.toDataStreamResponse();
+};
+```
+
+This file is doing a few key things:
+
+1. It sets up our OpenAI client with an API key.
+2. It defines different AI model configurations in the `variants` object.
+3. The `POST` function handles incoming chat requests.
+
+When a request comes in, it:
+
+-   Extracts the messages from the request body
+-   Selects a variant (currently hardcoded to 'variant1')
+-   Uses the OpenAI API to generate a response
+-   Streams the response back to the client
+
+The `streamText` function part of some utilities provided by Vercel's AI SDK, which helps deal with real-time streaming of the AI's responses and other AI-related tasks.
```

Comment:
This sentence is maybe missing a word?

Reply:
Maybe: The streamText function, part of Vercel’s AI SDK, assists with the real-time streaming of AI responses and other AI-related tasks.